tests/test_matrixsampling.py

```diff
@@ -0,0 +1,281 @@
+                assert approx(np.abs(np.linalg.det(m)), 1)
+                if det:
+                    assert approx(np.linalg.det(m), 1)
+                assert isinstance(m, MathArray)
+
+def test_orthogonal():
+    # Test shape, orthogonality, determinant, python2 error, MathArray
+    if six.PY2:
+        with raises(NotImplementedError, match='This feature requires newer versions of '
+                                               'numpy and scipy than are available.'):
+            matrices = OrthogonalMatrices(unitdet=False)
+            matrices.gen_sample()
+        with raises(NotImplementedError, match='This feature requires newer versions of '
+                                               'numpy and scipy than are available.'):
+            matrices = OrthogonalMatrices(unitdet=True)
+            matrices.gen_sample()
+    else:
+        # These are the doctests
+        matrices = OrthogonalMatrices()
+        assert matrices.gen_sample().shape == (2, 2)
+
+        matrices = OrthogonalMatrices(dimension=4)
+        assert matrices.gen_sample().shape == (4, 4)
+
+        matrices = OrthogonalMatrices(unitdet=True)
+        assert within_tolerance(np.linalg.det(matrices.gen_sample()), 1, 1e-14)
+
+        matrices = OrthogonalMatrices(unitdet=False)
+        m = matrices.gen_sample()
+        assert (within_tolerance(np.linalg.det(m), 1, 1e-14)
+                or within_tolerance(np.linalg.det(m), -1, 1e-14))
+
+        matrices = OrthogonalMatrices(unitdet=True)
+        m = matrices.gen_sample()
+        assert within_tolerance(m * np.conjugate(np.transpose(m)), MathArray(np.eye(2)), 1e-14)
+
+        matrices = OrthogonalMatrices(unitdet=False)
+        m = matrices.gen_sample()
+        assert within_tolerance(m * np.conjugate(np.transpose(m)), MathArray(np.eye(2)), 1e-14)
+
+        shapes = tuple(range(2, 5))
+        dets = (True, False)
+
+        # More general testing
+        for shape in shapes:
+            for det in dets:
+                matrices = matrices = OrthogonalMatrices(dimension=shape, unitdet=det)
+                m = matrices.gen_sample()
+                assert m.shape == (shape, shape)
+                assert np.array_equal(np.conj(m), m)
+                assert within_tolerance(m * np.transpose(m), MathArray(np.eye(shape)), 1e-14)
+                assert (approx(np.abs(np.linalg.det(m)), 1)
+                        or approx(np.abs(np.linalg.det(m)), -1))
+                if det:
+                    assert approx(np.linalg.det(m), 1)
+                assert isinstance(m, MathArray)
+
+def test_square_matrices():
+    # Test shape, real/complex, norm, symmetry, traceless, det, MathArray
+    shapes = tuple(range(2, 5))
+    norms = ([2, 6], [1 - 1e-12, 1 + 1e-12])
+    symmetries = (None, 'diagonal', 'symmetric', 'antisymmetric', 'hermitian', 'antihermitian')
+    traceless_opts = (True, False)
+    dets = (None, 0, 1)
+    complexes = (True, False)
+
+    for det in dets:
```

Comment:
If you want to reduce the indentation a bit,

```python
for (det, traceless, symmetry) in product(dets, traceless_opts, symmetries):
    ...
```
would work, right? Also below with `shape, norm, comp`

Reply:
Good point. I've also cleaned up the arguments.